src/Altinn.Apps/AppFrontend/react/altinn-app-frontend/src/components/base/CheckboxesContainerComponent.tsx

```diff
@@ -67,95 +68,46 @@ const useStyles = makeStyles({
-  React.useEffect(() => {
-    ref.current = value.slice();
-  });
 
-  return ref.current;
-}
+const emptyList: undefined[] = []; // constant for reference stability
 
 export const CheckboxContainerComponent = (props: ICheckboxContainerProps) => {
   const classes = useStyles(props);
-  const apiOptions = useAppSelector(state => state.optionState.options[props.optionsId]);
-  const options = apiOptions || props.options || [];
-  const [selected, setSelected] = React.useState([]);
-  const prevSelected: any = usePrevious(selected);
+  const apiOptions: IOption[] = useAppSelector(state => state.optionState.options[props.optionsId]);
+  const options = apiOptions || props.options || emptyList;
   const checkBoxesIsRow: boolean = options.length <= 2;
 
-  React.useEffect(() => {
-    returnState();
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [options]);
+  const selected = React.useMemo(() => {
+    return props.formData.simpleBinding.split(',').map(v => options.find(o => o.value == v)).filter(o => !!o) ?? emptyList;
+  }, [props.formData, options])
 
+  // Implement preselected functionality
   React.useEffect(() => {
-    returnState();
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [props.formData?.simpleBinding]);
-
-  const returnState = () => {
+    const preselectedOption = options?.[props.preselectedOptionIndex]?.value;
     if (
-      !props.formData?.simpleBinding &&
-      props.preselectedOptionIndex >= 0 &&
-      options &&
-      props.preselectedOptionIndex < options.length
+      !selected &&
+      preselectedOption
     ) {
-      const preSelected: string[] = [];
-      preSelected[props.preselectedOptionIndex] =
-        options[props.preselectedOptionIndex].value;
-      props.handleDataChange(preSelected[props.preselectedOptionIndex]);
-      setSelected(preSelected);
-    } else {
-      setSelected(props.formData?.simpleBinding ? props.formData.simpleBinding.split(',') : []);
+      props.handleDataChange(preselectedOption);
     }
-  };
-
-  const onDataChanged = (event: React.ChangeEvent<HTMLInputElement>) => {
-    const newSelected: any = selected.slice();
-    const index = newSelected.indexOf(event.target.name);
+    // This effect should only run when new options are loaded
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [options]);
 
-    if (index >= 0) {
-      newSelected[index] = '';
-    } else {
-      newSelected.push(event.target.name);
-    }
-    const filtered = newSelected.filter((element: string) => !!element);
-    props.handleFocusUpdate(props.id);
-    props.handleDataChange(selectedHasValues(filtered) ? filtered.join() : '');
-  };
 
-  const handleOnBlur = () => {
-    props.handleDataChange(props.formData?.simpleBinding ?? '');
-  };
 
-  const selectedHasValues = (select: string[]): boolean => {
-    return select.some((element) => element !== '');
-  };
+  const onDataChanged = (event: React.ChangeEvent<HTMLInputElement>) => {
```

Comment:
It would be good if this handler was renamed to `handleDataChanged`, as it is reacting to an event, not triggering it. I know we are not entirely consistent on these naming conventions in the codebase, but I am trying to improve that :) 

```suggestion
  const handleDataChanged = (event: React.ChangeEvent<HTMLInputElement>) => {
```

Reply:
Another issue is that we already have a function `props.handleDataChange` in this component. It would be somewhat confusing to have two functions with only one `d` in difference.